module/x/peggy/keeper/batch.go

```diff
@@ -71,17 +71,28 @@ func (k Keeper) OutgoingTxBatchExecuted(ctx sdk.Context, tokenContract string, n
 	})
 
 	// Delete batch since it is finished
-	k.deleteBatch(ctx, *b)
+	k.DeleteBatch(ctx, *b)
 	return nil
 }
 
-func (k Keeper) storeBatch(ctx sdk.Context, batch *types.OutgoingTxBatch) {
+// StoreBatch stores a transaction batch
+func (k Keeper) StoreBatch(ctx sdk.Context, batch *types.OutgoingTxBatch) {
 	store := ctx.KVStore(k.storeKey)
+	// set the current block height when storing the batch
+	batch.Block = uint64(ctx.BlockHeight())
 	key := types.GetOutgoingTxBatchKey(batch.TokenContract, batch.BatchNonce)
 	store.Set(key, k.cdc.MustMarshalBinaryBare(batch))
 }
 
-func (k Keeper) deleteBatch(ctx sdk.Context, batch types.OutgoingTxBatch) {
+// StoreBatchUnsafe stores a transaction batch w/o setting the height
```

Comment:
q: why would one use this method instead of the above one?

Reply:
When we are storing batches during normal operation we want to set the height at which the batch is stored on the batch for checking slashing. If, during tests we want to set some specific batch, or during InitGenesis we want to set batches with some height then we need the unsafe one.

Reply:
the long form explanation could go in the comment to avoid future questions. But it's not a big deal either way.